server/polar/exceptions.py

```diff
@@ -3,29 +3,33 @@
 IntegrityError = sqlalchemy.IntegrityError
 
 
-class AuthenticationRequired(Exception):
-    """Attempted action requires authentication."""
+class PolarError(Exception):
+    """
+    Base exception class for all errors raised by Polar.
 
+    A custom exception handler for FastAPI takes care
+    of catching and returning a proper HTTP error from them.
 
-class NotPermitted(Exception):
-    ...
-
-
-class InvalidPlatform(Exception):
-    ...
+    Args:
+        message: The error message that'll be displayed to the user.
+        status_code: The status code of the HTTP response. Defaults to 400.
+    """
 
-
-class InvalidRequest(Exception):
-    ...
+    def __init__(self, message: str, status_code: int = 400) -> None:
+        super().__init__(message)
+        self.message = message
+        self.status_code = status_code
 
 
-class StripeError(Exception):
-    ...
+class NotPermitted(PolarError):
+    def __init__(self, message: str = "Not permitted", status_code: int = 403) -> None:
```

Comment:
For convenience, those exceptions define a default message and status code, but this can be overridden when raising it if needed.